lib/galaxy_test/api/test_histories.py

```diff
@@ -140,6 +140,56 @@ def test_index_query(self):
         assert len(index_response) == 1
         assert index_response[0]["name"] == expected_history_name
 
+    def test_index_views(self):
+        # Make sure there is at least one history
+        self._create_history(f"TestHistoryForViews_{uuid4()}")["id"]
+        # By default the view is summary
+        index_response = self._get("histories").json()
+        for history in index_response:
+            assert "state" not in history
+
+        # Change the view to detailed
+        index_response = self._get("histories?view=detailed").json()
+        for history in index_response:
+            assert "state" in history
+
+        # Expect only specific keys
+        expected_keys = ["name"]
+        unexpected_keys = ["id", "deleted", "state"]
```

Comment:
I think the `HistoryMinimal` model here:

https://github.com/galaxyproject/galaxy/blob/8fcb994d07acae6107803942fd154d7658de4055/lib/galaxy/schema/schema.py#L1207

is forcing us always to serialize the `model_class`, `id`, and `user_id` (which might be a good thing?) the question is this is not fully compatible with what the `keys` serialization option was doing before.

`/api/histories?keys=name` would have returned in the past:
```json
[
  {
    "name": "Unnamed history"
  }
]

```

now it returns:
```json
[
  {
    "model_class": "History",
    "id": null,
    "user_id": null,
    "name": "Unnamed history"
  }
]
```

Is that ok? or should we always force these 3 attributes to be "properly" serialized when using "keys"?